its/src/test/java/com/sonar/it/scanner/msbuild/utils/ScannerCommand.java

```diff
@@ -0,0 +1,185 @@
+    return new ScannerCommand(Step.begin, classifier, token, projectDir, projectKey)
+      .setProperty("sonar.projectBaseDir", projectDir.toAbsolutePath().toString())
+      // Default values provided by Orchestrator to ScannerForMSBuild in adjustedProperties
+      .setProperty("sonar.scm.disabled", "true")
+      .setProperty("sonar.branch.autoconfig.disabled", "true");
+  }
+
+  public static ScannerCommand createEndStep(ScannerClassifier classifier, String token, Path projectDir) {
+    return new ScannerCommand(Step.end, classifier, token, projectDir, null);
+  }
+
+  public ScannerCommand setProperty(String key, @Nullable String value) {
+    if (value == null) {
+      this.properties.remove(key);
+    } else {
+      this.properties.put(key, value);
+    }
+    return this;
+  }
+
+  public ScannerCommand setDebugLogs(boolean verbose) {
+    return setProperty("sonar.verbose", Boolean.toString(verbose));
+  }
+
+  public ScannerCommand setEnvironmentVariable(String name, String value) {
+    if (value == null) {
+      environment.remove(name);
+    } else {
+      environment.put(name, value);
+    }
+    return this;
+  }
+
+  @Deprecated
+  public ScannerCommand addArgument(String argument) {
+    // ToDo: Remove in SCAN4NET-201.
+    if (argument == "begin") {
+      step = Step.begin;
+    } else if (argument == "end") {
+      step = Step.end;
+    } else {
+      throw new IllegalArgumentException();
+    }
+
+    return this;
+  }
+
+  @Deprecated
+  public ScannerCommand setProjectKey(String projectKey) {
+    // ToDo: Remove in SCAN4NET-201.
+    this.projectKey = projectKey;
+    return this;
+  }
+
+  @Deprecated
+  public ScannerCommand setProjectName(String name) {
+    // ToDo: Remove in SCAN4NET-201.
+    // It's no-op in-place replacement for now.
+    return this;
+  }
+
+  @Deprecated
+  public ScannerCommand setProjectVersion(String version) {
+    // ToDo: Remove in SCAN4NET-201.
+    // It's no-op in-place replacement for now.
+    return this;
+  }
+
+  @Deprecated
+  public ScannerCommand setProjectDir(File dir) {
+    // ToDo: Remove in SCAN4NET-201.
+    // It's no-op in-place replacement for now.
+    return this;
+  }
+
+  @Deprecated
+  public ScannerCommand setScannerVersion(String version) {
+    // ToDo: Remove in SCAN4NET-201.
+    // It's no-op in-place replacement for now.
+    return this;
+  }
+
+  @Deprecated
+  public ScannerCommand setUseDotNetCore(boolean value) {
+    // ToDo: Remove in SCAN4NET-201.
+    // It's no-op in-place replacement for now.
+    return this;
+  }
+
+  public BuildResult execute(Orchestrator orchestrator) {
+    var command = createCommand(orchestrator);
+    var result = new BuildResult();
+    LOG.info("Command line: {}", command.toCommandLine());
+    result.addStatus(CommandExecutor.create().execute(command, new StreamConsumer.Pipe(result.getLogsWriter()), Constants.COMMAND_TIMEOUT));
```

Comment:
Is `ORCHESTRATOR.executeBuild` obsolete with this PR? If so, why wasn't it removed just yet?

Reply:
That's a good point. I have two usages left that I should get rid of:
* testHelpMessage
* testHelp

They both seem to test the same things. I've created SCAN4NET-294 to clean that up, and to use this new class there, as I will need to add a new `step.None` there.